### ISSUE TITLE
Add Changes Since Milestone 1 and Future Plans section to report

### ISSUE THREAD

**Issue by EmreBatuhan** [bounswe/bounswe2023group4#487]
### Problem Definition

Read the requested information and fill the report accordingly.

**Comment by enfurars** [comment]
Good work here. Changes between milestone 1 and internal milestone should be added too. Other than that it is nice.

**Comment by EmreBatuhan** [comment]
Changes Since Milestone 1 and Future Plans

Overview

We mostly focused on polls and profiles since milestone 1. Poll view, poll creation were implemented for different types of polls. Also profile view, profile updation were implemented. Some poll and profile features were added like poll tags an profile badges. We also covered some of the errors that were made in milestone 1. Thanks to these changes our application has the main functionality that our application is supposed to have. The user can customize their profile and vote and create polls according to their preferences.
For our future plans, we will improve poll UI as the data displayed by it is not obvious from the user perspective. New poll UI will make the application easier to use and more fun to explore. We want to implement poll closing for all poll types. This will complete the poll life cycle. We also want to implement moderator system since we want moderators to prevent invalid and inappropriate polls. We will also add annotations for polls which will make the polls more interactable. We also plan to add topic points and we will implement leaderboard which will compare the topic points of users and reward them with badges according to their placement in the leaderboard. This way we will have a reward system for the users that have predicted in our polls.

Below we have included our detailed additions to the project by subteams.

Frontend Changelog
Vote Poll Page added
Create Poll Page added
Authentication Page Validation
Leaderboard Page
Point Backend Integration added
Profile Page added
Edit Profile Page added
Feed Page added
Search Bar UI added
Google Sign in added.
Responsiveness to Poll pages
Responsiveness to Authentication pages
Flaws
Moderation Page UI is not implemented

Backend Changelog
We have finalized google auth 
Implemented creating, seeing and voting discrete and continuous polls endpoints.
Image upload and view
Mock badges
Point system and poll closing is implemented
Profile maintenance, viewing and update of profiles
Poll closing and point
Improved error management
Email verification
Auth validation
Automatic Poll tagging
Poll service tests
Profile service tests
Authorization service tests
Flaws
More exhaustive tests can be written

Mobile Changelog
Profile page
Feed page
Create poll page
Leaderboard ui with mock data
Auth page validation
Google sign in
Removed mock data
Bug fixes (auth middleware fix, some crashes)
App bar
Drawer integration
DateTime (ISO8601) format support
Flaws:
Poll UI should be improved
Poll buttons' actions should be implemented.
Profile page is not completed we don't support some features like edit profile or badges.
Leaderboard ui will be improved when we connect it to the backend.
Drawer and main app looks differrent we may adjust their colors.
Test coverage should be increased.
